modules/actor/actor-wfrp4e.js

```diff
@@ -952,31 +950,32 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
     // e.g. the players can actually test to avoid an effect, instead of the GM doing it
     async applyEffect({effectUuids=[], effectData=[], messageId}={})
     {
-        let owningUser = game.wfrp4e.utility.getActiveDocumentOwner(this);
-
         if (typeof effectUuids == "string")
         {
             effectUuids = [effectUuids];
         }
 
-        if (owningUser?.id == game.user.id)
+        for (let uuid of effectUuids)
         {
-            for (let uuid of effectUuids)
-            {
-                let effect = fromUuidSync(uuid);
-                let message = game.messages.get(messageId);
-                await ActiveEffect.implementation.create(effect.convertToApplied(message?.getTest()), {parent: this, message : message?.id});
-            }
-            for(let data of effectData)
+            let effect = fromUuidSync(uuid);
+            let message = game.messages.get(messageId);
+            let data = effect.convertToApplied(message?.getTest());
+            effectData.push(data);
+        }
+
+        for (let data of effectData) 
+        {
+            let owningUser = game.wfrp4e.utility.getActiveDocumentOwner(this);
+            if (owningUser?.id == game.user.id)
             {
                 await ActiveEffect.implementation.create(data, {parent: this, message : messageId});
+            }   
+            else 
+            {
+                await game.wfrp4e.socket.executeOnOwnerAndWait(this, "applyEffect", {effectUuids : [], effectData : [data], actorUuid : this.uuid, messageId});
```

Comment:
What is the point of this change? Reevaluating `owningUser` multiple times is pointless as it won't change.

Reply:
i guess i can move it to top. The goal of this change is to normalize how effect data is being passed, so that we could await the sockets always. 
I was also thinking about having a flag in ```getActiveDocumentOwner(this, gmOnly);``` and another flag 'Run as GM' in effect configuration. This way i could define an effect that should be run only as GM, but since i dont see a valid point for that yet - i only wanted to prepare this function and make it fully awaitable.

Reply:
Unless there is some pressing need I am unaware of, I don't think this change is needed. The only thing I can think of is from within a script, but that can also simply use `await game.wfrp4e.socket.executeOnOwnerAndWait(this, "applyEffect")` itself